plugins/jobs/src/js/data/JobModel.ts

```diff
@@ -524,17 +527,98 @@ const fieldResolvers = {
   }
 };
 
+// Sort and filtering
+function sortJobs(
+  jobs: Job[],
+  sortBy: SortOption = "ID",
+  sortDirection: SortDirection = "ASC"
+): Job[] {
+  jobs.sort((a, b) => {
+    let result;
+
+    switch (sortBy) {
+      case "ID":
+        result = sortJobById(a, b);
+        break;
+      case "STATUS":
+        result = sortJobByStatus(a, b);
+        break;
+      case "LAST_RUN":
+        result = sortJobByLastRun(a, b);
+        break;
+      default:
+        result = 0;
+    }
+
+    const direction = sortDirection === "ASC" ? 1 : -1;
+    return result * direction;
+  });
+
+  return jobs;
+}
+
+function sortJobById(a: Job, b: Job): number {
+  return a.id.localeCompare(b.id);
+}
+
+function sortJobByStatus(a: Job, b: Job): number {
+  return (
+    JobStates[a.scheduleStatus].sortOrder -
+    JobStates[b.scheduleStatus].sortOrder
+  );
+}
+
+function sortJobByLastRun(a: Job, b: Job): number {
+  return (
+    JobStatus[a.lastRunStatus.status].sortOrder -
+    JobStatus[b.lastRunStatus.status].sortOrder
+  );
+}
+
+function filterJobs(
+  jobs: MetronomeClient.JobResponse[],
+  filter: string | null
+): MetronomeClient.JobResponse[] {
+  if (filter === null) {
+    return jobs;
+  }
+
+  return jobs.filter(({ id }) => id.indexOf(filter) !== -1);
+}
+
 export const resolvers = ({
+  fetchJobs,
   fetchJobDetail,
   pollingInterval
 }: ResolverArgs): IResolvers => ({
   Query: {
     jobs(
-      _obj = {},
-      _args: GeneralArgs = {},
+      _parent = {},
+      args: GeneralArgs = {},
       _context = {}
-    ): Observable<JobConnection | null> {
-      return Observable.of(null);
+    ): Observable<Job[] | null> {
+      const {
+        sortBy = "ID",
+        sortDirection = "ASC",
+        filter = null
+      } = args as JobsQueryArgs;
+
+      const pollingInterval$ = Observable.timer(0, pollingInterval);
+      const responses$ = pollingInterval$.switchMap(fetchJobs);
+
+      return (
+        responses$
+          .map(jobs => filterJobs(jobs, filter))
+          // N+1 request madness
+          .map(jobs => jobs.map(job => fetchJobDetail(job.id)))
+          .switchMap(obs => Observable.combineLatest(obs))
+          .map(requests => requests.map(typeResolvers.Job))
+          .map(jobs =>
+            //TODO Why typescript can't figure out that these value is always
+            // defined?
```

Comment:
because `sortBy` and `sortDirection` aren't typed, this should work 🤔 
```
+      const {
+        sortBy: SortOption = "ID",
+        sortDirection: SortDirection = "ASC",
+        filter: string | null = null
+      } = args;
```

Reply:
You are right. I solved that this way before, and then I rebased and lost that.

Reply:
We have the `JobsQueryArgs`, they are generated from GraphQL and well typed. You would need to write a [user defined type guard](https://github.com/basarat/typescript-book/blob/master/docs/types/typeGuard.md#user-defined-type-guards) to "cast" it into the right type. I would prefer the generated code over the handwritten TBH, I don't know how you feel about this @juliangieseke

Reply:
generally yes, but it looks like its already messed up, not sure if we are able to re-generate 😢

Reply:
I actually was not able to type on destruction, I made the item non nullable.